tests/unit/log_printer_test.py

```diff
@@ -40,11 +41,30 @@ def reader(*args, **kwargs):
 
         self.assertIn(glyph, output)
 
+    def test_notail(self):
+        def reader(*args, **kwargs):
+            for count in xrange(3):
+                yield "keep going\n"
+                time.sleep(1)
```

Comment:
The underlying queue has a 0.1 second timeout on polling for data, so I didn't find a better way than a sleep, although I could reduce it to 0.25s to speed up the test.



> On Jan 6, 2015, at 2:37 PM, Daniel Nephin <notifications@github.com> wrote:
> 
> In tests/unit/log_printer_test.py:
> 
> > @@ -40,11 +41,30 @@ def reader(*args, **kwargs):
> >  
> >          self.assertIn(glyph, output)
> >  
> > +    def test_notail(self):
> > +        def reader(*args, **kwargs):
> > +            for count in xrange(3):
> > +                yield "keep going\n"
> > +                time.sleep(1)
> I always try to avoid any sleep() in tests. It slows things down unnecessarily.
> 
> Is this actually required for the tests to pass?
> 
> —
> Reply to this email directly or view it on GitHub.
>